prov/verbs/src/verbs_eq.c

```diff
@@ -1346,7 +1388,7 @@ vrb_eq_sread(struct fid_eq *eq_fid, uint32_t *event,
 	eq = container_of(eq_fid, struct vrb_eq, eq_fid.fid);
 
 	while (1) {
-		ret = vrb_eq_read(eq_fid, event, buf, len, flags);
+		ret = vrb_eq_read_internal(eq_fid, timeout, event, buf, len, flags);
```

Comment:
The timeout should not be passed through.  Waiting will occur on the eq->epollfd, which has the async_fd attached.  If an async event occurs, the epollfd will be signaled.